config/clusters/2i2c/demo.values.yaml

```diff
@@ -20,7 +20,14 @@ jupyterhub:
           url: https://2i2c.org
   hub:
     config:
+      JupyterHub:
+        authenticator_class: cilogon
       Authenticator:
         # We do not define allowed_users here since only usernames matching this regex will be allowed to login into the hub.
         # Ref: https://jupyterhub.readthedocs.io/en/stable/api/auth.html#jupyterhub.auth.Authenticator.username_pattern
         username_pattern: '^(.+@2i2c\.org|.+@rmbl\.org|deployment-service-check)$'
+      CILogonOAuthenticator:
+        oauth_callback_url: https://demo.2i2c.cloud/hub/oauth_callback
+        username_claim: email
+        allowed_idps:
+          - "2i2c.org"
```

Comment:
@yuvipanda, do we still want to provide access to `mbl.org` domains? If so, we should add that to the list of idps I think and get rid of `username_pattern` which does the same thing?

Reply:
@GeorgianaElena so I'm actually a bit confused about allowed_idps. Does it just check email domain? How will it deal with me having a GitHub account with an email set to somehacker@rmbl.org?

Reply:
I'm going to merge it as it is now and we can address it later I believe 🚀 TY!

Reply:
@GeorgianaElena great! am testing it again

Reply:
I missed your message @yuvipanda, sorry.

> Does it just check email domain? How will it deal with me having a GitHub account with an email set to [somehacker@rmbl.org](mailto:somehacker@rmbl.org)?

If the `usesrname_claim` is `email` and you login though GitHub, then yes, I believe it will check anyting after `@` to be in the `allowed_idps` list if such a list exists. 

https://github.com/jupyterhub/oauthenticator/blob/8c1cf0c616d5e19d1f93238673f17db2daf57fa0/oauthenticator/cilogon.py#L193-L199